File: countwords.py
# -*- coding: utf-8 -*-
import os

my_file = open(os.path.join(os.path.abspath('.'), 'text.txt'),'r')

# 读单词
my_list=[] 

for item in my_file.read().split(" "):
  my_list.append(item)

list_len = len(my_list)

print my_list

# 定义计频函数
def frequency(word):
  fre = 0
  n = 0
  while n < list_len:
    if word == my_list[n]:
      fre += 1
    n += 1
  return fre

# 建立单词-频数字典
my_dict={}

for item in my_list:
  my_dict[item] = frequency(item)

print my_dict

# 频数排序
fre_list = my_dict.values()
fre_list.sort()
fre_list = list(set(fre_list))
print fre_list

# 按频数读key，写新txt
# 这里最好用 path 来自动序列化绝对路径 @guo-yu
new_file = open(os.path.join(os.path.abspath('.'), 'output.txt'), 'w')

for x in fre_list:
  for item in my_dict.keys():
    if my_dict[item] == x:
      new_file.write(str(item)+"\n")

new_file.close()
my_file.close()

File: node/countwords.js
var fs = require('fs')
var path = require('path')

var srcFile = fs.readFileSync(path.resolve('../', 'text.txt'))
var srcFileWords = srcFile.toString().split(' ') // assume we cound get correct words by this way
var wordMap = {}
var finalText = ''

// count and collection them to a Object/Map/Dict
srcFileWords.forEach(function(word){
  if (!wordMap[word])
    wordMap[word] = 0

  wordMap[word] ++
})

// That's do this:
finalText = Object.keys(wordMap)
  // Create a Array with word itself and it's count
  .map(function(word){
    return {
      word: word,
      count: wordMap[word]
    }
  })
  // Sort by object.count( words number)
  .sort(function(a, b){
    return a.count < b.count
  })
  // Create a string that what we wanted
  .map(function(item){
    return item.word + ' shows ' + item.count + ' times'
  }).join('\n')

// Write that result to disk:
fs.writeFileSync('output.csv', finalText)